test/hooks.test.js

```diff
@@ -2561,6 +2561,80 @@ test('preSerialization hooks should support encapsulation', t => {
   })
 })
 
+test('onEncapsulation hook should be called / 1', t => {
```

Comment:
Could you add a test with an `async` function?

Reply:
I think it is not needed, the hook is a sync function, and `async` plugins behave in the same way of classic callback plugins :)

Reply:
You have right, there is not `next` or `done` functions 👍